<launch>

  <arg name="orb_voc_path" default = "$(find trajectory_servoing)/../ORB_Data/"/>
  <arg name="path_slam_config" default = "$(find trajectory_servoing)/configs/slam_configs/"/>
  <arg name="path_slam_config_file" default = "multisense_camera.yaml"/>
  <arg name="path_track_logging" default = "/DATA/IVAL/traj_servoing_benchmark/test/slam_files/test"/>
  <arg name="path_map_logging" default = "/DATA/IVAL/traj_servoing_benchmark/test/slam_map/test_Map"/>
  <arg name="num_good_feature" default = "180" />
  <arg name="do_rectify" default = "false" />
  <arg name="do_vis" default = "false" />

  <!-- call GF-ORB-SLAM for vision-based state estimation -->
  <!-- Note DON'T enable on-screen output for GF-SLAM! otherwise it will lead to  --> 
  <node pkg="gf_orb_slam2" type="StereoFeatures" name="visual_slam" args="$(arg orb_voc_path)/ORBvoc.bin $(arg path_slam_config)/$(arg path_slam_config_file) $(arg num_good_feature) $(arg do_rectify) $(arg do_vis) /multisense_sl/camera/left/image_raw /multisense_sl/camera/right/image_raw $(arg path_track_logging) $(arg path_map_logging)">
  </node>

</launch>

